src/leaderboard/messages/data/delete.rs

```diff
@@ -0,0 +1,45 @@
+use crate::leaderboard::messages::MomentoRequest;
+use crate::utils::prep_leaderboard_request_with_timeout;
+use crate::{Leaderboard, MomentoResult};
+
+/// Request to delete a leaderboard
+pub struct DeleteRequest {}
+
+impl DeleteRequest {
+    /// Constructs a new `DeleteRequest`.
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for DeleteRequest {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MomentoRequest for DeleteRequest {
+    type Response = DeleteResponse;
+
+    async fn send(self, leaderboard: &Leaderboard) -> MomentoResult<Self::Response> {
+        let cache_name = leaderboard.cache_name();
+        let request = prep_leaderboard_request_with_timeout(
```

Comment:
yeah, "timeout" word is correct here, but needs fixed 2 lines down

Reply:
Addressed in https://github.com/momentohq/client-sdk-rust/pull/438#discussion_r1943405912